src/main/java/io/gravitee/resource/oauth2/generic/OAuth2GenericResource.java

```diff
@@ -148,8 +148,8 @@ protected void doStart() throws Exception {
                 .setDefaultPort(authorizationServerPort)
                 .setDefaultHost(authorizationServerUrl.getHost())
                 .setMaxPoolSize(configuration().getHttpClientOptions().getMaxConcurrentConnections())
-                .setIdleTimeout(60)
-                .setConnectTimeout(10000);
+                .setIdleTimeout(configuration().getHttpClientOptions().getIdleTimeout())
```

Comment:
with my previous remark, `idleTimeout` will be in milliseconds.
I would recommend to add a `.setIdleTimeoutUnit(TimeUnit.MILLISECONDS)` to make it explicit

Reply:
it might require to clamp the value of the long to not exceed Integer.MAX